src/Elmish.WPF/Merge.fs

```diff
@@ -15,22 +16,58 @@ type DuplicateIdException (sourceOrTarget: SourceOrTarget, index1: int, index2:
   member this.Index2 = index2
   member this.Id = id
 
+type internal CollectionTarget<'a> =
+  { GetLength: unit -> int
+    GetAt: int -> 'a
+    Append: 'a -> unit
+    InsertAt: int * 'a -> unit
+    ReplaceAt: int * 'a -> unit
+    RemoveAt: int -> unit
+    Move: int * int -> unit
+    Clear: unit -> unit
+    Enumerate: unit -> 'a seq
+    BoxedCollection: unit -> obj }
+
+module internal CollectionTarget =
+  let map (f: 'a -> 'b) (f': 'b -> 'a) (ct: CollectionTarget<'a>) : CollectionTarget<'b> =
+    { GetLength = ct.GetLength
+      GetAt = ct.GetAt >> f
+      Append = f' >> ct.Append
+      InsertAt = Pair.mapAll id f' >> ct.InsertAt
+      ReplaceAt = Pair.mapAll id f' >> ct.ReplaceAt
+      RemoveAt = ct.RemoveAt
+      Move = ct.Move
+      Clear = ct.Clear
+      Enumerate = ct.Enumerate >> Seq.map f
+      BoxedCollection = ct.BoxedCollection }
+
+  let create (oc: ObservableCollection<'a>) =
```

Comment:
Everywhere we use this function (outside tests, which are easily refactored), we are first transforming a `Seq` to an `ObservableCollection`. Would it be a good idea to extract the transformation here and just let `create` take a `Seq`?

Reply:
The problem is with the functions below. Seq is read-only. The functions below actually modify the observable collection.

Also ultimately we return the `BoxedCollection` property when we call `Get` on a property that was built with this, and that needs to be the actual collection (not a transformed version of it).